src/Migrations/M025ReplaceDeprecatedColorsFromContent.php

```diff
@@ -0,0 +1,95 @@
+<?php
+
+namespace P4\MasterTheme\Migrations;
+
+use P4\MasterTheme\MigrationRecord;
+use P4\MasterTheme\MigrationScript;
+
+/**
+ * Replace the deprecated pallete colors from content
+ */
+class M025ReplaceDeprecatedColorsFromContent extends MigrationScript
+{
+    /**
+     * Perform the actual migration.
+     *
+     * @param MigrationRecord $record Information on the execution, can be used to add logs.
+     * phpcs:disable SlevomatCodingStandard.Functions.UnusedParameter -- interface implementation
+     */
+    protected static function execute(MigrationRecord $record): void
+    {
+        global $wpdb;
+
+        $colors = array(
+            [ 'from' => 'grey-80', 'to' => 'grey-900' ],
+            [ 'from' => 'grey-60', 'to' => 'grey-800' ],
+            [ 'from' => 'grey-40', 'to' => 'grey-600' ],
+            [ 'from' => 'grey-20', 'to' => 'grey-500' ],
+            [ 'from' => 'grey-10', 'to' => 'grey-200' ],
+            [ 'from' => 'grey-05', 'to' => 'grey-100' ],
+            [ 'from' => 'gp-green', 'to' => 'green-500' ],
+            [ 'from' => 'dark-blue', 'to' => 'dark-green-800' ],
+            [ 'from' => 'orange-hover', 'to' => 'green-500' ],
+            [ 'from' => 'grey', 'to' => 'grey-900' ],
+            [ 'from' => 'blue', 'to' => 'green-800' ],
+        );
+
+        foreach ($colors as $color) {
+            $from = $color['from'];
+            $to = $color['to'];
+
+            $from_bg = sprintf("has-%s-background-color", $color['from']);
+            $to_bg = sprintf("has-%s-background-color", $color['to']);
+            $from_color = sprintf("has-%s-color", $color['from']);
+            $to_color = sprintf("has-%s-color", $color['to']);
+
+            echo sprintf("Replace all the occurrences of '%s' with '%s'\n", $color['from'], $color['to']);
+
+            $sql = '
+                SELECT
+                    ID,
+                    post_parent,
+                    MAX(ID) as latest_post,
+                    count(ID) as total_revisions
+                FROM wp_posts
+                WHERE (
+                    post_content LIKE "%%%1$s%%"
+                    OR post_content LIKE "%%%2$s%%"
+                )
+                GROUP BY post_parent
+            ';
```

Comment:
Testing on international, I think there's an issue with the request, because of WP handling of `post_parent`:
- if page_type = 'revision', page_parent is the ID of the original post
- if page_type = 'page' or 'post' or that kind of stuff, page_parent is an actual parent in the structure of the site.

so you don't get children pages in this request. I think you could make a request with `post_type` <> 'revision' to get the original IDs, and then get the last revision of those. But i'm not even sure you need to modify revisions, as the last revision is a copy of the current page. Just changing the select query should work
```sql
            SELECT
                ID
            FROM wp_posts
            WHERE
            post_type <> "revision" AND (
                post_content LIKE "%%%1$s%%"
                OR post_content LIKE "%%%2$s%%"
            )
```

aaaand I realize that the last modification I asked now prevents updating blocks attributes `backgroundColor` and `textColor`, so we should add more REPLACE statements with attributes
```php
        $from_bg_attr = sprintf("\"backgroundColor\":\"%s\"", $color['from']);
        $to_bg_attr = sprintf("\"backgroundColor\":\"%s\"", $color['to']);
// etc...
```